kernel/kernel-api/src/main/java/io/delta/kernel/internal/actions/GenerateIcebergCompatActionUtils.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal.actions;
+
+import static io.delta.kernel.internal.data.TransactionStateRow.*;
+import static io.delta.kernel.internal.util.Preconditions.checkArgument;
+
+import io.delta.kernel.Transaction;
+import io.delta.kernel.data.Row;
+import io.delta.kernel.engine.Engine;
+import io.delta.kernel.exceptions.KernelException;
+import io.delta.kernel.expressions.Literal;
+import io.delta.kernel.internal.TableConfig;
+import io.delta.kernel.internal.data.TransactionStateRow;
+import io.delta.kernel.internal.fs.Path;
+import io.delta.kernel.internal.icebergcompat.IcebergCompatV2MetadataValidatorAndUpdater;
+import io.delta.kernel.utils.CloseableIterable;
+import io.delta.kernel.utils.DataFileStatus;
+import java.net.URI;
+import java.util.Map;
+
+/** Utilities to convert Iceberg add/removes to Delta Kernel add/removes */
+public final class GenerateIcebergCompatActionUtils {
+
+  /**
+   * Create an add action {@link Row} that can be passed to {@link Transaction#commit(Engine,
+   * CloseableIterable)} from an Iceberg add.
+   *
+   * @param transactionState the transaction state from the built transaction
+   * @param fileStatus the file status to create the add with (contains path, time, size, and stats)
+   * @param partitionValues the partition values for the add
+   * @param dataChange whether or not the add constitutes a dataChange (i.e. append vs. compaction)
+   * @return add action row that can be committed to the transaction
+   * @throws UnsupportedOperationException if icebergWriterCompatV1 is not enabled
+   * @throws UnsupportedOperationException if maxRetries != 0 in the transaction
+   * @throws KernelException if stats are not present (required for icebergCompatV2)
+   * @throws UnsupportedOperationException if the table is partitioned (currently unsupported)
+   */
+  public static Row generateIcebergCompatWriterV1AddAction(
+      Row transactionState,
+      DataFileStatus fileStatus,
+      Map<String, Literal> partitionValues,
+      boolean dataChange) {
+    Map<String, String> config = getConfiguration(transactionState);
```

Comment:
"config" is a big generic -- which config is this? this is the Metadata Configuration (a better name is Table Properties) of the transaction, right?

So perhaps `transactionTableProperties` or something like that?

Reply:
yeah I'm good with `transactionTableProperties`. We do use `configuration` everywhere throughout the code though -- probably mainly because the field is `metadata.configuration` :/

Reply:
actually the getter method is even `getConfiguration`-- I think we use `configuration` pretty much everywhere throughout the code except when it's user facing like SQL, txnBuilder API.

I'd prefer to leave this for now as it's more standard with everything else in the code-base. Whether all those instances should be changed is a different matter

Reply:
👍